hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/Utilities.java

```diff
@@ -0,0 +1,257 @@
+        assertThat(cluster.location()).isEqualTo(parameters.location());
+        assertThat(cluster.getTags()).isEqualTo(parameters.tags());
+        assertThat(cluster.properties().connectivityEndpoints()).has(new Condition<List<? extends ConnectivityEndpoint>>() {
+            @Override
+            public boolean matches(List<? extends ConnectivityEndpoint> connectivityEndpoints) {
+                for (ConnectivityEndpoint connectivityEndpoint : connectivityEndpoints) {
+                    if (connectivityEndpoint.name().equals("HTTPS")) {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        });
+        assertThat(cluster.properties().connectivityEndpoints()).has(new Condition<List<? extends ConnectivityEndpoint>>() {
+            @Override
+            public boolean matches(List<? extends ConnectivityEndpoint> connectivityEndpoints) {
+                for (ConnectivityEndpoint connectivityEndpoint : connectivityEndpoints) {
+                    if (connectivityEndpoint.name().equals("SSH")) {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        });
+        assertThat(cluster.properties().osType()).isEqualTo(parameters.properties().osType());
+        assertThat(cluster.properties().errors()).isNull();
+        assertThat(cluster.properties().provisioningState()).isEqualTo(HDInsightClusterProvisioningState.SUCCEEDED);
+        assertThat(cluster.properties().clusterDefinition().kind())
+            .isEqualTo(parameters.properties().clusterDefinition().kind());
+        assertThat(cluster.properties().clusterVersion()).startsWith(parameters.properties().clusterVersion());
+        assertThat(cluster.properties().clusterDefinition().configurations()).isNull();
+    }
+
+    public static void validateHttpSettings(
+        Map<String, String> httpSettings,
+        String expectedUserName,
+        String expectedUserPassword) {
+        assertThat(httpSettings).isNotNull();
+        assertThat(httpSettings.get("restAuthCredential.isEnabled")).isEqualTo("true");
+        assertThat(httpSettings.get("restAuthCredential.username")).isEqualTo(expectedUserName);
+        assertThat(httpSettings.get("restAuthCredential.password")).isEqualTo(expectedUserPassword);
+    }
+
+    public static String generateRandomClusterName(String prefix) {
+        return generateRandomClusterName(prefix, 35);
+    }
+
+    public static String generateRandomClusterName(String prefix, int maxLen) {
+        return TestBase.generateRandomResourceName(prefix, maxLen);
+    }
+
+    public static Role findRoleByName(ClusterInner cluster, final String roleName) {
+        return Iterables.find(
+            cluster.properties()
+                .computeProfile()
+                .roles(),
+            new Predicate<Role>() {

```

Comment:
The lambda function may work here for more simple

Reply:
Lambda expression is not available in Java 7. So I have to use anonymous class instead.